src/ace/ari_text/parsemod.py

```diff
@@ -361,7 +363,16 @@ def p_ampair(p):
 
 def p_structlist_join(p):
     'structlist : structlist structpair'
-    p[0] = p[1] | p[2]  # merge dicts
+    merged = p[1].copy()  # Start with left side
+    
+    # Check for duplicates while merging dicts
+    for key, value in p[2].items():
+        if key in merged:
+            LOGGER.error("Multiple nonce definitions found")
```

Comment:
It would be better to use a generic message here such as "Parameter list has duplicate key X" since nonce is specific to `EXECSET` / `RPTSET` but not applicable to `TBL`.